lib/builder.js

```
@@ -296,6 +306,14 @@ function getPluginName(tree) {
 
 exports.getDescription = getDescription
 function getDescription (tree) {
+  if (typeof tree !== 'string') {
+    try {
```

What is this protecting us against? I don’t think we want to swallow any/all errors here.

I think the swap to broccoli-node-info should address the originally reported issue properly without the try / catch...

---

https://github.com/broccolijs/broccoli-node-info

> This function throws a broccoliNodeInfo.InvalidNodeError when called with

* a plain string node (while still supported by the Broccoli Builder, they are deprecated in favor of broccoli-source),
* a node using the old .read/.rebuild API (see docs/broccoli-1-0-plugin-api.md), or
* some other object that isn't a node.

Seem like 2 could be a problem

---

ya, sounds good, lets check for the specific error and only swallow that one

---

🆗

---

Done